parquet-hadoop/src/main/java/org/apache/parquet/hadoop/DictionaryPageReader.java

```diff
@@ -48,12 +50,23 @@
 
   private final ParquetFileReader reader;
   private final Map<String, ColumnChunkMetaData> columns;
-  private final Map<String, DictionaryPage> cache = new HashMap<String, DictionaryPage>();
+  private final Map<String, Optional<DictionaryPage>> cache;
```

Comment:
I find a bit weird to have an optional as the value.

Reply:
Sure can.